python/cudf/cudf/core/reshape.py

```diff
@@ -983,8 +983,13 @@ def pivot(data, index=None, columns=None, values=None):
         raise ValueError("Duplicate index-column pairs found. Cannot reshape.")
 
     result = _pivot(values, index, columns)
+
+    # MultiIndex to Index
     if not values_is_list:
-        result.columns = result.columns.droplevel(0)
+        column_name = result._data.level_names[1]
+        table_columns = [multi_col[1] for multi_col in result._data.names]
+        result.columns = cudf.Index(data=table_columns, name=column_name)
```

Comment:
We should really avoid touching the `.columns` attribute altogether. And we should _definitely_ avoid setting it  to a cuDF object as that will force a device-to-host copy. See the logic in the columns [setter](https://github.com/rapidsai/cudf/blob/127d57485445001e5d0c9fdc20fd8e380d9a410d/python/cudf/cudf/core/dataframe.py#L2244) method.

I think there's a bug in the `ColumnAccessor` droplevels function that makes it difficult to use here. Happy  to chat more about this offline, and we can see if including a fix  for that in this PR is appropriate.

Reply:
I think I found the issue with `droplevel`. `names` is a cached property under `ColumnAccessor` but we expect it to change when a level is dropped. I've done a [fix](https://github.com/rapidsai/cudf/pull/11538/commits/788acff0404a293db685545a83c8b0002081cca2#diff-3541948bd042d45b52e62ae79b29555eae9dc2d00e060f058818c0ebb0c0b44dR619) for it and things seem to be working fine now. Let me know if something else needs to be changed.